public/components/homepage.js

```diff
@@ -0,0 +1,176 @@
+        return;
+      }
+      searchWarning.innerHTML = "";
+      let category = document.getElementById('filter-category-dropdown').value;
+      let location = document.getElementById('filter-location-dropdown').value;
+      let query = `/search?keyword=${searchName}`;
+        // + (category ? `&category=${category}` : '')
+        // + (location ? `&location=${location}` : '');
+      ajaxRequest('GET', query, { ajax : true }, {}, renderFreelancers);
+    }
+  }
+  render() {
+    return (
+      <div id="search-container">
+        <input id="search-name" placeholder="Who?" onKeyDown={this.searchRequest} />
+        <button id="search-btn" onClick={this.searchRequest}>Search</button>
+        <div id="search-warning"></div>
+      </div>
+    );
+  }
+}
+
+class FiltersContainer extends React.Component {
+  render() {
+    let categories = [];
+    for (let i = 0; i < this.props.categories.length; ++i) {
+      let name = this.props.categories[i].categoryName;
+      categories.push(<option value={name} key={i}>{name}</option>);
+    }
+    let locations = [];
+    for (let i = 0; i < this.props.locations.length; ++i) {
+      locations.push(<option value={this.props.locations[i]} key={i}>{this.props.locations[i]}</option>);
+    }
+    return (<div id="filters-container">
+      <div id="filters">
+        <div id="filter-category">
+          <span>Category: </span>
+          <select id="filter-category-dropdown" defaultValue="">
+            <option value="">Anything</option>
+            {categories}
+          </select>
+        </div>
+        <div id="filter-location">
+          <span>Location: </span>
+          <select id="filter-location-dropdown" defaultValue="">
+            <option value="">Anywhere</option>
+            {locations}
+          </select>
+        </div>
+      </div>
+    </div>);
+  }
+}
+
+class FreelancerCreateBtn extends React.Component {
+  redirectFreelancerForm() {
+    document.location = '/freelance/new';
+  }
+  render () {
+    return (<button id="freelancer-create-btn" onClick={this.redirectFreelancerForm}>Add a new freelancer</button>);
+  }
+}
+
+class FreelancerName extends React.Component {
+  render () {
+    return (<h2>{this.props.first} {this.props.last}</h2>);
+  }
+}
+
+class FreelancerCard extends React.Component {
+  redirectFreelancer(freelancer) {
+    return function() {
+      document.location = `/freelance/${freelancer.props._id}`;
+    }
+  }
+  render () {
+    return (
+      <div className="freelancer-card" onClick={this.redirectFreelancer(this)}>
+        <div className="freelancer-card-picture-placeholder"><img src={this.props.urlPicture} /></div>
+        <div className="freelancer-card-info">
+          <h1>{this.props.title}</h1>
+          <FreelancerName first={this.props.first} last={this.props.last}/>
+          <span>{this.props.category}</span>
+          <span>Average Score: {this.props.avgScore} in 5</span>
```

Comment:
I'd write `{this.props.avgScore || -}/5`, similar to freelancer-view. (`|| -` for the case in which there's no score.)